elements/lisk-p2p/test/integration/p2p.ts

```diff
@@ -1763,4 +1765,83 @@ describe('Integration tests for P2P library', () => {
 			});
 		});
 	});
+
+	describe('Network with peer inbound eviction protection for connectTime enabled', () => {
+		const NETWORK_PEER_COUNT_WITH_LIMIT = 10;
+		const MAX_INBOUND_CONNECTIONS = 3;
+		const POPULATOR_INTERVAL_WITH_LIMIT = 150;
+		beforeEach(async () => {
+			p2pNodeList = [...new Array(NETWORK_PEER_COUNT_WITH_LIMIT).keys()].map(
+				index => {
+					// Each node will have the previous node in the sequence as a seed peer except the first node.
+					const seedPeers = [
+						{
+							ipAddress: '127.0.0.1',
+							wsPort:
+								NETWORK_START_PORT +
+								((index + 1) % NETWORK_PEER_COUNT_WITH_LIMIT),
+						},
+					];
+
+					const nodePort = NETWORK_START_PORT + index;
+					return new P2P({
+						connectTimeout: 100,
+						ackTimeout: 5000,
+						seedPeers,
+						wsEngine: 'ws',
+						populatorInterval: POPULATOR_INTERVAL_WITH_LIMIT,
+						maxOutboundConnections: MAX_INBOUND_CONNECTIONS,
+						maxInboundConnections: MAX_INBOUND_CONNECTIONS,
+						latencyProtectionRatio: 0,
+						productivityProtectionRatio: 0,
+						longevityProtectionRatio: 0.5,
+						nodeInfo: {
+							wsPort: nodePort,
+							nethash:
+								'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
+							version: '1.0.1',
+							protocolVersion: '1.0.1',
+							minVersion: '1.0.0',
+							os: platform(),
+							height: 0,
+							broadhash:
+								'2768b267ae621a9ed3b3034e2e8a1bed40895c621bbb1bbd613d92b9d24e54b5',
+							nonce: `O2wTkjqplHII${nodePort}`,
+						},
+					});
+				},
+			);
+
+			// Start nodes incrementally to make inbound eviction behavior predictable
+			p2pNodeList.forEach(async p2p => {
+				await wait(500);
+				p2p.start();
+			});
+			await wait(1500);
+		});
+
+		afterEach(async () => {
+			await Promise.all(
+				p2pNodeList
+					.filter(p2p => p2p.isActive)
+					.map(async p2p => await p2p.stop()),
+			);
+			await wait(100);
+		});
+
+		describe('Inbound peer evictions', () => {
```

Comment:
I would strongly suggest to add at least two cases more to cover the three different variables.

Reply:
It's hard to test for `latency` in a predictable manner without getting varying results in this integration test.. Also the same for `responseRate`. Since it is integration test I tried to only manipulate the p2p configurations and node start times, without calling internal functions directly to test the behavior. The only metric that is easily tested for in this case is `connectTime`.

Reply:
What do you think about at least creating a test skeleton that you're able to think of now that it's a fresh topic? Then we can cover later. Otherwise we will forget.